### DISCUSSION TITLE
TypeError: argument of type 'NoneType' is not iterable

### DISCUSSION THREAD

**Discussion by Abdurrafey-Siddiqui** [openai/whisper#696] [Q&A] [answered]
When I try to import whisper I get this error:

```if` '/' in name or '\\' in name:
TypeError: argument of type 'NoneType' is not iterable```

**Comment by Abdurrafey-Siddiqui** [comment]
nvm, I forced reinstalled whisper and it worked.

**Reply by bruhcode234** [reply]
mine still show that error even though i force reinstalled it, do you find other way to fix that?

**Reply by bruhcode234** [reply]
nvm, i think i installed the wrong packages. should've used pip install git+https://github.com/openai/whisper.git not pip install whisper

**Reply by Abdurrafey-Siddiqui** [reply]
use the command on the read me:

pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git

**Reply by HoussamZak** [reply]
I am still facing the same issue despite running the solution mentioned previously

**Reply by palhanobrazil** [reply]
after all that i get this message "ERROR: No matching distribution found for torch"

**Reply by TommyTunaToro** [reply]
After doing force reinstall, I am still getting the same error message: "    if '/' in name or '\\' in name:
                                                                                                           TypeError: argument of type 'NoneType' is not iterable"
I have downgraded from 3.11 to 3.10

**Reply by MustafaCQN** [reply, marked as answer]
For who has the same error and can't fix:

**You seem installed wrong package as `pip install whisper`**
**But you need to install this package `pip install openai-whisper`**

so you should first uninstall whisper then install openai-whisper
1. `pip uninstall whisper`
2. `pip install openai-whisper`

**Reply by GavinAren** [reply]
Worked!

**Reply by ibtehajali67** [reply]
thank you, its worked

**Reply by mozekang** [reply]
thank you a lot

**Reply by shreyassapkal** [reply]
Thank you, It's working

**Reply by facenl** [reply]
Thank you, It's working！！！

**Comment by rafaelmca** [comment]
I use Python 3.12 and it shows this same error.

<img width="610" alt="WHISPER_CODE" src="https://github.com/openai/whisper/assets/113183319/d6f3ecab-f2d8-4d50-8cde-d36c7d236a10">

**Comment by rodrigomv29** [comment]
I believe the error comes from installing `whisper` instead of `openai-whisper`
